tests/integration/CMakeLists.txt

```diff
@@ -45,14 +45,14 @@ foreach(data_dir "ring" "ring_gap")
     file(COPY
          "${CMAKE_CURRENT_SOURCE_DIR}/${data_dir}/out.dat.ref"
          DESTINATION
-         "${CMAKE_CURRENT_SOURCE_DIR}/${data_dir}_${test_suffix}/")
+         "${CMAKE_CURRENT_BINARY_DIR}/${data_dir}_${test_suffix}/")
   endforeach()
 endforeach()
 
 file(COPY
      "${CMAKE_CURRENT_SOURCE_DIR}/ring/out.dat.ref"
      DESTINATION
-     "${CMAKE_CURRENT_SOURCE_DIR}/ring_spike_buffer/")
+     "${CMAKE_CURRENT_BINARY_DIR}/ring_spike_buffer/")
 
```

Comment:
I am surprised that these were created in `CMAKE_CURRENT_SOURCE_DIR`!

Reply:
for a very very long time :)

Reply:
@alkino since you asked: https://github.com/BlueBrain/CoreNeuron/blob/735bfa582c630e55fae68a865d703b3c06209b14/tests/integration/integration_test.sh.in#L31

the `ring_spike_buffer` directory has a `out.dat.ref` file which is overwritten.. now you don't overwrite it anymore.. not quite sure what the author "to blame" was quite intending but @pramodk can tell us and we decide.

Reply:
> the ring_spike_buffer directory has a out.dat.ref file which is overwritten

* If you look carefully, it didn't i.e. note `${data_dir}/out.dat.ref` vs `${data_dir}_${test_suffix}/out.dat.ref`
* What was happening is that we have single source input data (e.g. `ring/out.dat.ref`) that we wanted to use for different tests (e.g. ring test but using multi-send spike exchange method or binqueu for queuing or AoS/SoA layout). So for simplicity of integration  tests, the reference result was just copied to different directories. 

I will push some fixes. We have missed some stuff in #179.